### DISCUSSION TITLE
几万字符，rec模型训练rec_r34_vd_none_bilstm_ctc.yml，初始lr设置为多少？后期很难改善时，lr设置为多少？

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#14669] [Q&A]
Global:
  use_gpu: true
  epoch_num: 72
  log_smooth_window: 20
  print_batch_step: 100
  save_model_dir: 
  save_epoch_step: 3
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 2000]
  cal_metric_during_train: false #true
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_words_en/word_10.png
  # for data or label process
  character_dict_path: 
  max_text_length: 25
  infer_mode: False
  use_space_char: true
  save_res_path: ./

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.0005
  regularizer:
    name: 'L2'
    factor: 0

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: ResNet
    layers: 34
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 256
  Head:
    name: CTCHead
    fc_decay: 0

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

**Comment by GreatV** [comment]
在 `rec_r34_vd_none_bilstm_ctc.yml` 配置文件中，初始学习率 (`learning_rate`) 被设置为 **0.0005**，这与提供的参考文档一致。

如果在训练的后期模型提升变得困难，可以尝试以下方法调整学习率 (`lr`)：

1. **降低学习率**：
   - 逐步减少学习率，例如调整为 **0.0001** 或 **0.00005**，这有助于细化优化过程，防止参数剧烈波动。
   - 可以使用 **学习率衰减策略**（如 `ReduceLROnPlateau` 或 `StepLR`）来自动调整学习率。

2. **使用预训练模型继续训练**：
   - 如果训练效果停滞，可以尝试加载之前的最佳模型权重，并以更低的学习率（如 **0.00005**）继续训练。

3. **调整优化器参数**：
   - Adam 自适应优化器通常效果较好，但如果优化停滞，可以尝试 **更换优化器**（如 SGD + Momentum）或调整 `beta1` 和 `beta2` 参数。

综上，建议：
- 初始 `lr`: **0.0005**
- 训练后期难以改善时，可逐步降低 `lr` 至 **0.0001** 或 **0.00005** 进行微调。

如果仍然无法提升，可以检查数据质量、模型架构或数据增强策略。

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*
